Newsletters: add 248 (2023-04-26)

harding opened the pull request:
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Bitcoin StackExchange section @bitschmidty

Xekyo commented on _topics/en/transaction-origin-privacy.md:
```diff
@@ -0,0 +1,106 @@
+- [v2 P2P transport][topic v2 p2p transport]: this protocol will provide
+  native encryption of transaction and other data for Bitcoin programs.
+
+- [Dandelion][topic dandelion]: this protocol will allow a node which
+  did not originate a transaction to be the first to publicly broadcast
+  it, potentially making it much more difficult to determine the actual
+  origin.  Dandelion will be much more effective if it is combined with
+  encryption, either naively or through anonymity networks.
```
I thought that Dandelion has basically been given up on, since the DoS vectors it introduces could not be efficiently mitigated. From what I understand, all the concerns described by sdaftuar in this [BSE post](https://bitcoin.stackexchange.com/a/81504/5406) still apply

harding replied:
There was discussion in bitcoin-core-dev IRC recently about "[one-hop dandelion](https://www.erisian.com.au/bitcoin-core-dev/log-2023-04-22.html#l-76)", which (I just learned) means having the origination node announce the transaction over an anonymity network (e.g. Tor) but not store it in its own mempool until it receives it from the P2P network.  In other words, to announce the transaction using an ephemeral identity and then act like any other node when it later sees that transaction.

Should that properly be called "dandelion"?  I dunno.  But it seems like dandelion is still a concept that several developers are thinking about, so I think it's beneficial to list it as something for new learners.

Xekyo reviewed (commented):
Looks good so far, just left one comment on Dandelion

harding commented on _topics/en/transaction-origin-privacy.md:
```diff
@@ -0,0 +1,106 @@
+## Optional.  Each entry requires "title" and "url".  May also use "feature:
+## true" to bold entry and "date"
+optech_mentions:
+  - title: "Bitcoin Core #18861 won't reply to with txes it hasn't yet announced to the requesting peer"
```
We're (self) limited to 100 characters here.  I'll fix the typo and s/txes/transactions/, but I don't have space for mentioning `getdata`.  Happy to take other suggested rephrasings within 100 characters, though.

LarryRuane commented on _posts/en/newsletters/2023-04-26-newsletter.md:
```diff
@@ -0,0 +1,147 @@
+    first.  On the mailing list, the only [reply][harding mempool] as of
+    this writing noted that lightweight clients that connect to their
+    own trusted node can currently use BIP35 and BIP37 to learn about
+    unconfirmed transactions in a much more bandwidth efficient manner
```
```suggestion
    unconfirmed transactions in a much more bandwidth-efficient manner
```

LarryRuane commented on _posts/en/newsletters/2023-04-26-newsletter.md:
```diff
@@ -0,0 +1,147 @@
+  time][], which would only allow updating a gossip message once per
+  block (approximately every 10 minutes on average).  Although the PR
+  notes that sending slower updates involves tradeoffs, it updates LDK
+  to waiting about 10 minutes before broadcasting a channel disabled
```
```suggestion
  to wait about 10 minutes before broadcasting a channel disabled
```

LarryRuane commented on _topics/en/transaction-origin-privacy.md:
```diff
@@ -0,0 +1,106 @@
+  **Transaction origin privacy** is the ability to hide the origin of a
+  transaction from network surveillance.
+---
+If an eavesdropper can see every transaction relayed by every node, they
```
```suggestion
If eavesdroppers can see every transaction relayed by every node, they
```
so the "they" is grammatically correct -- although I'm aware of the more recent usage, I'm old school, free free to ignore.

LarryRuane commented on _posts/en/newsletters/2023-04-26-newsletter.md:
```diff
@@ -0,0 +1,147 @@
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [How many sigops are in the invalid block 783426?]({{bse}}117837)
+  Vojtěch Strnad provided a script which iterates through all transactions in a
```
```suggestion
  Vojtěch Strnad provided a script that iterates through all transactions in a
```

harding commented on _topics/en/transaction-origin-privacy.md:
```diff
@@ -0,0 +1,106 @@
+    url: /en/newsletters/2019/09/18/#txprobe-discovering-bitcoin-s-network-topology-using-orphan-transactions
+
+  - title: "Bitcoin Core #14897 introduces a semi-random order when requesting transactions"
+    url: /en/newsletters/2019/02/12/#bitcoin-core-14897
```
We did mention both of those in subsequent newsletters:

- https://bitcoinops.org/en/newsletters/2019/04/23/#bitcoin-core-15839
- https://bitcoinops.org/en/newsletters/2019/06/19/#bitcoin-core-15834

I've added 15834 to the topic as it also links to 15839 (I don't want to add both of them since they're not directly about transaction origin privacy; I do want to add one of them, though, as it's conceivable that another implementation could stumble upon the same problem).

harding commented on _posts/en/newsletters/2023-04-26-newsletter.md:
```diff
@@ -0,0 +1,147 @@
+- [Core Lightning 23.05rc1][] is a release candidate for the next
+  version of this LN node.
+
+- Bitcoin Core 25.0RC1 FIXME:harding
```
Note, dropping the Bitcoin Core RC.  As of this writing, it hasn't been uploaded to bitcoincore.org and I don't think it's urgent that we announce it this week.

harding commented:
Made edits (or left replies) to all feedback, except this one for @bitschmidty : https://github.com/bitcoinops/bitcoinops.github.io/pull/1094#discussion_r1176997179  (thanks reviewers @Xekyo @jonatack @LarryRuane !).  Reviewed @bitschmidty section (looks great!), added lede, updated releases/RCs, and added topic entries.

Thanks everyone!

bitschmidty commented:
Pushed a change in one the topic mentions text for Transaction origin privacy, squashed.

bitschmidty commented:
Thanks to @harding for authoring this week and @Xekyo @LarryRuane @jonatack for the review!
